Tools/ardupilotwaf/mavgen.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python
+# encoding: utf-8
+# (c) Siddharth Bharat Purohit, 3DRobotics Inc.
+
+"""
+The **mavgen.py** program is a code generator which creates mavlink header files.
+"""
+
+from waflib import Task
+from waflib.TaskGen import extension
+import os
+
+class mavgen(Task.Task):
+	"""generate mavlink header files"""
+	color   = 'BLUE'
+	run_str = '${MAVGEN} --lang=C --wire-protocol=1.0 --output=generated_src/include/mavlink/v1.0/ ${SRC[0].abspath()}'
+
+@extension('.xml')
+def process_mavgen(self, node):
+	"""
+	Create a mavgen task
+	"""
+	tsk = self.create_task('mavgen', node, [])
+
+def configure(cfg):
+	"""
+	setup environment for mavlink header generator
+	"""
+	env = cfg.env
+	cfg.env.env = dict(os.environ)
```

Comment:
@bugobliterator 

The issues I see with that approach is:
 - you're overwriting a possible previously set system environment
 - I don't think it's a good idea to set PYTHONPATH at configure time:
   1. Other tools relying on PYTHONPATH might have problems since you are overwriting the variable. You should try to prepend to it instead, however look at (ii).
   2.  Even if you prepend to it, that will have effect for all build, not only mavlink headers generation.

So I think that you should set PYTHONPATH at run time.